### ISSUE TITLE
Leading underscore doesn't disable error on unused type parameter

### ISSUE THREAD

**Issue by RyanCavanaugh** [Microsoft/TypeScript#18420]
From  #18370

This reports an unused local error on `_T`, which we shouldn't do because it has a leading underscore
```ts
type Foo<_T> = { };
```

**Comment by mhegazy** [comment]
PRs welcomed
